### ISSUE TITLE
Debug Backplane Fixes (RevB errata)

### ISSUE THREAD

**Issue by adkinsjd** [lab11/signpost#23]
-  [x] Switch module 1 and 0 programming silkscreen (light is still correct)

**Comment by adkinsjd** [comment]
- [x] Probably should put default off resistors on the debug LEDs

**Comment by adkinsjd** [comment]
- [x] Nit - put the programming signals in order
 - [x] Move silkscreen out to accommodate a knurled knob

**Comment by adkinsjd** [comment]
- [x] A big 5V warning. maybe  a protection circuit

**Comment by ppannuto** [comment]
Moving from #26 here:

 - [x] Add LED for fake radio module.

**Comment by ppannuto** [comment]
- [x] Deal with USB Hub Reset problem
![image](https://cloud.githubusercontent.com/assets/339422/22951767/69ad599a-f2d8-11e6-8211-c6c734981d7f.png)

**Comment by brghena** [comment]
## USB Hub Reset

Reboot: Hub still good
Power cycle (with ten seconds physically powered off): Hub still good

I'm guessing just asserting BackplaneReset at boot will be sufficient

**Comment by bradjc** [comment]
- [x] Doesn't say "Signpost"
- [x] Module 0, Module 1, Controller font are not the same size.
- [x] "MEM" -> "Storage"
- [x] What is the empty programming switch position for?
- [x] Move the JTAG header to the top of the board.

**Comment by brghena** [comment]
- [x] No easily accessible ground post for probing (Oscilloscope alligator clippy thing is too big for any of the 0.1" debug headers)

**Comment by adkinsjd** [comment]
Literally every screw? Or the 4 gnd testpoints at various spots on the
board?

**Comment by brghena** [comment]
Sorry, first message was unclear. I edited it to include the real intent, which is the ground connections on oscilloscope leads, which are too big and would short to other pins on the 0.1" headers and cannot latch on to round-headed screws.

**Comment by ppannuto** [comment]
I'm a fan of the ground bar-like thing. Draw a rectangular poly of exposed ground metal along the edge so that leads can just clip to the pcb.

**Comment by adkinsjd** [comment]
You should unscrew one of the feet for now

**Comment by bradjc** [comment]
- [x] Mod0 and Mod1 debug headers should be identical.

**Comment by brghena** [comment]
- [x] Enable programming of modules even without a valid controller app

Currently, when you just plug a module in to the debug backplane, you can't program it unless the controller has disabled isolation for that module. One solution to this would be powering modules explicitly even when the programming switch is set to it.

**Comment by bradjc** [comment]
- [x] Bigger !RESET button switches.

**Comment by adkinsjd** [comment]
I am actually pretty strongly against this unless you can enable and
disable it strictly when the jlink is actually flashing. I think that a
module staying on when the controller wants it off would be more
problematic in the long run, and I don't think requiring turning the switch
to be selecting none of the boards is viable, as you would inevitably
forget.

There is probably a jlink signal that could switch on/off for the duration
of programming that could make this happen though.

**Comment by ppannuto** [comment]
Yeah, the whole flashing / dev story needs a little work. Currently we have
the annoying problem where flashing one quite often tends to leave other
boards in the "i2c lines stuck low until reset" state that the SAM4L likes
to get into.

The big hammer is obviously to put all boards into reset during a flash,
power on the target module forcefully, and then take things out of reset -
but that could be obnoxious in practice. I'd rather understand this buggy
i2c state

**Comment by adkinsjd** [comment]
The buggy I2C state is a separate issue, which I worry we will never be
able to fix because it's a chip bug (although the first thing I would try
is adding a few extra jlink resets at the end of the flashing script).

**Comment by ppannuto** [comment]
Yeah.. adding jlink resets won't help. It's _other_ SAM4Ls that aren't
being flashed that tend to hold low it in my experience.

**Comment by adkinsjd** [comment]
Really? That's not what I've found. Usually if I hit the reset button on
the one I'm flashing it works.

**Comment by adkinsjd** [comment]
I just assumed something about pressing the button reset the chip
differently

**Comment by ppannuto** [comment]
Often is too strong. I'd say.. 75% of the time the one I flashed, maybe 25%
of the time another chip on the network somewhere. But at this point I'm
just in the habit of resetting all of them, so my empirical evidence isn't
great. Maybe I can unlearn that habit.

**Comment by bradjc** [comment]
My experience is basically 100% of the time it's the fault of the one I just flashed and that resetting the just flashed module fixes the issue.

**Comment by adkinsjd** [comment]
Have we tried just running JLinkExe and resetting a few times to see if
that fixes it?

**Comment by ppannuto** [comment]
- [x] Master reset button that resets everything

**Comment by ppannuto** [comment]
- [x] Another USB hub upstream of the FTDIs so that you can just plug in one USB cable

**Comment by nealjack** [comment]
- [ ] Radio Module clock unstable - gpios don't work on cold boot
    - > Neal Jackson [17:13] 
That was an issue I saw on my weird led flipped board - so I'm not sure it's an actual issue. I added a delay in the radio app that seemed to stabilize things
    - This was added to work around issue https://github.com/lab11/signpost/commit/ac68837df1fd405e7306dded32b567f29f427b06#diff-777998fb3b386dba9f5f4d2db1f9852cR119

**Comment by ppannuto** [comment]
- [x] Add GPIO extender to reset / monitor the USB hub. Set to address 3

**Comment by ppannuto** [comment]
- [x] power metering jumpers for modules/controller
 - [x] eeprom chip
